src/metabase/integrations/ldap.clj

```diff
@@ -183,10 +183,10 @@
    (with-connection find-user username))
 
   ([conn username]
-   (when-let [{:keys [dn], :as result} (search conn username)]
-     (let [{fname (keyword (ldap-attribute-firstname))
-            lname (keyword (ldap-attribute-lastname))
-            email (keyword (ldap-attribute-email))}    result]
+   (when-let [{:keys [dn], :as result} (into (empty {}) (for [[key value] (search conn username)] [(keyword (.toLowerCase (name key))) value]))]
+     (let [{fname (keyword (.toLowerCase ^String (setting/get :ldap-attribute-firstname)))
```

Comment:
I'm completely new to Clojure and I was trying a bunch of different things. This was the only way I could get it to work. I also wasn't fully satisfied with it since it felt like a hack and not as elegant as the previous code. 
There might be a much nicer solution to that which I just don't know of due to my lack of Clojure experience. I'm happy to receive and suggestions on how to improve this part. 
By also using the lower-case-en function here it already will look a bit better.

Reply:
Your "new Clojure" doesn't look bad at all. :slightly_smiling_face: 

You could simply wrap the setting function call in `lower-case-en`:

```clojure
(u/lower-case-en (ldap-attribute-firstname))
```

_However_, I just realized that there's an even better way, higher up the chain, where we can do the conversion: as setting getter functions!

```clojure
(defsetting ldap-attribute-firstname
  (deferred-tru "Attribute to use for the user''s first name. (usually ''givenName'')")
  :default "givenName"
  :getter (fn []
            (u/lower-case-en (setting/get-string :ldap-attribute-firstname))))
```

With that you can go back to using just the setting function like before, and you'll get a lower cased string back:

```clojure
(let [fname (keyword (ldap-attribute-firstname)) ...
```

Reply:
Done.